File: src/test/java/com/github/m50d/margatroid/MargatroidTest.java
package com.github.m50d.margatroid;

import static org.junit.Assert.*;
import org.junit.Test;

import com.github.m50d.margatroid.model.Value;
import com.github.m50d.margatroid.model.ast.Literal;

public class MargatroidTest {
	Margatroid interpreter = new Margatroid();

	@Test
	public void interpreter() {
		Value result = interpreter.run("[ + 1 2 3 ]");
		assertEquals(new Literal("6"), result);
	}
	
	@Test
	public void function() {
		Value result = interpreter.run("[ proc plus2 { x } { [ + 2 $x ] } ] [ plus2 3 ]");
		assertEquals(new Literal("5"), result);
	}

}

File: src/test/scala/com/github/m50d/margatroid/MargatroidTest.scala
package com.github.m50d.margatroid

import org.junit.Test
import org.junit.Assert._
import com.github.m50d.margatroid.model.ast.Literal

class MargatroidTest {
	val interpreter = new Margatroid()

	@Test
	def basicFunctionality() = {
		val result = interpreter.run("[ + 1 2 3 ]")
		assertEquals(new Literal("6"), result)
	}
	
	@Test
	def function() = {
		val result = interpreter.run("[ proc plus2 { x } { [ + 2 $x ] } ] [ plus2 3 ]")
		assertEquals(new Literal("5"), result)
	}
}